packages/internals/src/engine-commands/formatSchema.ts

```diff
@@ -1,52 +1,120 @@
+      }
+
+      const _schema = fs.readFileSync(_schemaPath, { encoding: 'utf8' })
+      return _schema
+    })
+    .exhaustive()
 
-  const options = {
-    env: {
-      RUST_BACKTRACE: process.env.RUST_BACKTRACE ?? '1',
-      ...(showColors ? { CLICOLOR_FORCE: '1' } : {}),
+  const formattedSchema = await handleFormatPanic(
+    () => {
+      // the only possible error here is a Rust panic
+      return formatWASM(schemaContent)
     },
-    maxBuffer: MAX_BUFFER,
-  } as execa.Options
+    { schemaPath, schema } as FormatSchemaParams,
+  )
 
-  let result
+  return formattedSchema
+}
 
-  if (process.env.FORCE_PANIC_PRISMA_FMT) {
-    result = await execa(prismaFmtPath, ['debug-panic'], options)
+function handleFormatPanic<T>(tryCb: () => T, { schemaPath, schema }: FormatSchemaParams) {
+  try {
+    return tryCb()
+  } catch (e: unknown) {
+    const wasmError = e as Error
+    throw new RustPanic(
+      /* message */ wasmError.message,
+      /* rustStack */ wasmError.stack || 'NO_BACKTRACE',
+      /* request */ '@prisma/prisma-fmt-wasm format',
+      ErrorArea.FMT_CLI,
+      schemaPath,
+      schema,
+    )
   }
+}
 
-  if (schemaPath) {
-    if (!fs.existsSync(schemaPath)) {
-      throw new Error(`Schema at ${schemaPath} does not exist.`)
-    }
-    result = await execa(prismaFmtPath, ['format', '-i', schemaPath], options)
-  } else if (schema) {
-    result = await execa(prismaFmtPath, ['format'], {
-      ...options,
-      input: schema,
-    })
+type DocumentUri = string
+
+type TextDocument = {
+  /**
+   * The associated URI for this document. Most documents have the __file__-scheme, indicating that they
+   * represent files on disk. However, some documents may have other schemes indicating that they are not
+   * available on disk.
+   */
+  readonly uri: DocumentUri
+}
+
+// Part of the LSP spec: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#documentFormattingParams
+// These are only the parts we are interested in.
+type DocumentFormattingParams = {
+  textDocument: TextDocument
+  options: {
+    // this is the only property currently considered by Rust, the rest are ignored but are needed for successfully unmarshaling the `DocumentFormattingParams`
+    // and be compatible with the LSP spec.
+    // The WASM formatter may fail silently on unmarshaling errors (a `warn!` macro is used in the Rust code, but that's not propagated to WASM land).
+    tabSize: number
+
+    insertSpaces: boolean
   }
+}
+
+const defaultDocumentFormattingParams: DocumentFormattingParams = {
+  textDocument: { uri: 'file:/dev/null' },
+  options: {
+    tabSize: 2,
+    insertSpaces: true,
+  },
+}
 
-  return result.stdout
+async function formatWASM(schema: string): Promise<string> {
+  const params: DocumentFormattingParams = defaultDocumentFormattingParams
+  const formattedSchema = prismaFmt.format(schema, JSON.stringify(params))
+  return Promise.resolve(formattedSchema)
 }
```

Comment:
`Promise.resolve(formattedSchema)` is to keep the existing return type?

We could return the Promise in the return line 59 instead?

```
return Promise.resolve(formatWASM(schemaContent))
```

Because it's a bit weird to do this in this function, which doesn't need it?

Reply:
True, this was needed before, and is thus just debt. Removed